x/incentives/spec/02_state.md

```diff
@@ -0,0 +1,58 @@
+<!--
+order: 2
+-->
+
+# State
+
+## Incentives management
+
+All the incentives that is going to be provided are locked into `IncentivePool` until it's released to the users after a specific period of time.
+
+### Pot
+
+A `Pot` is a unit of incentive distributer. It describes the the users who can get reward, and the amount of coins to be distribute, the distribution start time, and the number of epochs it provide incentive from the `Pot`.
+
+```go
+type DistrCondition struct {
+  LockType   uint64 // type of lock, ByLockDuration | ByLockTime
+  Denom  string // lock denom
+  Duration time.Duration // condition for lock duration, only valid if positive
+  Timestamp time.Time // condition for lock start time, not valid if unset value
```

Comment:
Is it simple to encode these as a protobuf oneof? (Since this really are LockType SpecificData)

Reply:
This struct is DistrCondition, and I think current struct is just the protobuf that describes lock type.